doc/v5.0/man3/MPI_File_set_atomicity.3
.\" -*- nroff -*-
.\" Copyright 2010 Cisco Systems, Inc.  All rights reserved.
.\" Copyright 2006-2008 Sun Microsystems, Inc.
.\" Copyright (c) 1996 Thinking Machines Corporation
.\" Copyright 2015-2016 Research Organization for Information Science
.\"                     and Technology (RIST). All rights reserved.
.\" Copyright (c) 2020      Google, LLC. All rights reserved.
.\" $COPYRIGHT$
.TH MPI_File_set_atomicity 3 "Sep 30, 2021" "5.0.0rc1" "Open MPI"
.SH NAME
\fBMPI_File_set_atomicity\fP \- Sets consistency semantics for data-access operations (collective).

.SH SYNTAX
.ft R
.nf
.SH C Syntax
.nf
#include <mpi.h>
int MPI_File_set_atomicity(MPI_File \fIfh\fP, int \fIflag\fP)

.fi
.SH Fortran Syntax
.nf
USE MPI
! or the older form: INCLUDE 'mpif.h'
MPI_FILE_SET_ATOMICITY(\fIFH\fP, \fIFLAG\fP, \fIIERROR\fP)
	INTEGER	\fIFH, FLAG, IERROR\fP

.fi
.SH Fortran 2008 Syntax
.nf
USE mpi_f08
MPI_File_set_atomicity(\fIfh\fP, \fIflag\fP, \fIierror\fP)
	TYPE(MPI_File), INTENT(IN) :: \fIfh\fP
	LOGICAL, INTENT(IN) :: \fIflag\fP
	INTEGER, OPTIONAL, INTENT(OUT) :: \fIierror\fP

.fi
.SH INPUT PARAMETERS
.ft R
.TP 1i
fh
File handle (handle).
.TP 1i
flag
\fBtrue\fP to enable atomic mode, \fBfalse\fP to enable nonatomic mode (boolean).

.SH OUTPUT PARAMETER
.ft R
.TP 1i
IERROR
Fortran only: Error status (integer).

.SH DESCRIPTION
.ft R
The consistency semantics for data-access operations using the set of
file handles created by one collective MPI_File_open is set by collectively
calling MPI_File_set_atomicity. All processes in the group must pass identical values for
.I fh
and
.I flag.
If
.I flag
is
.I true,
atomic mode is set; if
.I flag
is
.I false,
nonatomic mode is set.
.sp
The default value on a call to MPI_File_open in Open MPI is \fItrue\fP for jobs running on more than one node, \fIfalse\fP for jobs running on a single SMP. For more information, see the MPI-2 standard.

.SH ERRORS
Almost all MPI routines return an error value; C routines as the value of the function and Fortran routines in the last argument.
.sp
Before the error value is returned, the current MPI error handler is
called. For MPI I/O function errors, the default error handler is set to MPI_ERRORS_RETURN. The error handler may be changed with MPI_File_set_errhandler; the predefined error handler MPI_ERRORS_ARE_FATAL may be used to make I/O errors fatal. Note that MPI does not guarantee that an MPI program can continue past an error.



doc/v1.4/man3/MPI_File_set_atomicity.3.php
<?php
$topdir = "../../..";
$title = "MPI_File_set_atomicity(3) man page (version 1.4.5)";
$meta_desc = "Open MPI v1.4.5 man page: MPI_FILE_SET_ATOMICITY(3)";

include_once("$topdir/doc/nav.inc");
include_once("$topdir/includes/header.inc");
?>
<p> <a href="../">&laquo; Return to documentation listing</a></p>
  <!-- manual page source format generated by PolyglotMan v3.2, -->
<!-- available at http://polyglotman.sourceforge.net/ -->

<body bgcolor='white'>
<a href='#toc'>Table of Contents</a><p>

<h2><a name='sect0' href='#toc0'>Name</a></h2>
<b>MPI_File_set_atomicity</b> - Sets consistency semantics for data-access
operations (collective).
<p>
<h2><a name='sect1' href='#toc1'>Syntax</a></h2>
<br>
<pre>C Syntax
    #include &lt;mpi.h&gt;
    int MPI_File_set_atomicity(MPI_File fh, int flag)
</pre>
<h2><a name='sect2' href='#toc2'>Fortran Syntax</a></h2>
<br>
<pre>    INCLUDE &rsquo;mpif.h&rsquo;
    MPI_FILE_SET_ATOMICITY(FH, FLAG, IERROR)
    <tt> </tt>&nbsp;<tt> </tt>&nbsp;      INTEGER <tt> </tt>&nbsp;<tt> </tt>&nbsp;  FH, FLAG, IERROR
</pre>
<h2><a name='sect3' href='#toc3'>C++ Syntax</a></h2>
<br>
<pre>#include &lt;mpi.h&gt;
void MPI::File::Set_atomicity(bool flag)
</pre>
<h2><a name='sect4' href='#toc4'>Input Parameters</a></h2>

<dl>

<dt>fh </dt>
<dd>File handle (handle). </dd>

<dt>flag </dt>
<dd><b>true</b> to enable atomic mode,
<b>false</b> to enable nonatomic mode (boolean).
<p> </dd>
</dl>

<h2><a name='sect5' href='#toc5'>Output Parameter</a></h2>

<dl>

<dt>IERROR </dt>
<dd>Fortran
only: Error status (integer).
<p> </dd>
</dl>

<h2><a name='sect6' href='#toc6'>Description</a></h2>
The consistency semantics for
data-access operations using the set of file handles created by one collective
<a href="../man3/MPI_File_open.3.php">MPI_File_open</a> is set by collectively calling MPI_File_set_atomicity. All
processes in the group must pass identical values for  <i>fh</i>  and <i>flag.</i> If
 <i>flag</i>  is  <i>true,</i> atomic mode is set; if  <i>flag</i>  is  <i>false,</i> nonatomic mode
is set. <p>
The default value on a call to <a href="../man3/MPI_File_open.3.php">MPI_File_open</a> in Open MPI is <i>true</i>
for jobs running on more than one node, <i>false</i> for jobs running on a single
SMP. For more information, see the MPI-2 standard.
<p>
<h2><a name='sect7' href='#toc7'>Errors</a></h2>
Almost all MPI
routines return an error value; C routines as the value of the function
and Fortran routines in the last argument. C++ functions do not return errors.
If the default error handler is set to MPI::ERRORS_THROW_EXCEPTIONS, then
on error the C++ exception mechanism will be used to throw an MPI:Exception
object. <p>
Before the error value is returned, the current MPI error handler
is called. For MPI I/O function errors, the default error handler is set
to MPI_ERRORS_RETURN. The error handler may be changed with <a href="../man3/MPI_File_set_errhandler.3.php">MPI_File_set_errhandler</a>;
the predefined error handler MPI_ERRORS_ARE_FATAL may be used to make I/O
errors fatal. Note that MPI does not guarantee that an MPI program can continue
past an error.
<p> <p>

<hr><p>
<a name='toc'><b>Table of Contents</b></a><p>
<ul>
<li><a name='toc0' href='#sect0'>Name</a></li>
<li><a name='toc1' href='#sect1'>Syntax</a></li>
<li><a name='toc2' href='#sect2'>Fortran Syntax</a></li>
<li><a name='toc3' href='#sect3'>C++ Syntax</a></li>
<li><a name='toc4' href='#sect4'>Input Parameters</a></li>
<li><a name='toc5' href='#sect5'>Output Parameter</a></li>
<li><a name='toc6' href='#sect6'>Description</a></li>
<li><a name='toc7' href='#sect7'>Errors</a></li>
</ul>


<p> <a href="../">&laquo; Return to documentation listing</a></p>
<?php
include_once("$topdir/includes/footer.inc");
